Sources/Brave/Frontend/Sync/SyncSettingsTableViewController.swift

```diff
@@ -82,112 +140,101 @@ class SyncSettingsTableViewController: UIViewController, UITableViewDelegate, UI
-    case .otherDevice:
+    case .otherDeviceLeft:
       title = String(format: Strings.syncRemoveOtherDeviceTitle, deviceName)
       message = Strings.syncRemoveOtherDeviceMessage
       removeButtonName = Strings.removeDevice
+    case .syncChainDeleteConfirmation:
+      title = Strings.Sync.syncDeleteAccountAlertTitle
+      message = "\(Strings.Sync.syncDeleteAccountAlertDescriptionPart1).\n\n\(Strings.Sync.syncDeleteAccountAlertDescriptionPart2)"
+      removeButtonName = Strings.delete
+    case .syncChainDeleteError:
+      title = Strings.Sync.syncChainAccountDeletionErrorTitle
+      message = Strings.Sync.syncChainAccountDeletionErrorDescription
+      removeButtonName = Strings.OKString
     }
 
     guard let popupTitle = title, let popupMessage = message, let popupButtonName = removeButtonName else { fatalError() }
 
-    let popup = AlertPopupView(imageView: nil, title: popupTitle, message: popupMessage)
-    let fontSize: CGFloat = 15
+    let popup = AlertPopupView(title: popupTitle, message: popupMessage)
 
-    popup.addButton(title: Strings.cancelButtonTitle, fontSize: fontSize) { return .flyDown }
-    popup.addButton(title: popupButtonName, type: .destructive, fontSize: fontSize) {
+    popup.addButton(title: Strings.cancelButtonTitle) { return .flyDown }
+    popup.addButton(title: popupButtonName, type: .destructive) {
       switch type {
-      case .lastDeviceLeft, .currentDevice:
+      case .lastDeviceLeft, .currentDeviceLeft:
         self.syncAPI.leaveSyncGroup()
         self.navigationController?.popToRootViewController(animated: true)
-      case .otherDevice:
-        self.syncAPI.removeDeviceFromSyncGroup(deviceGuid: device.guid)
+      case .otherDeviceLeft:
+        if let guid = device?.guid {
+          self.syncAPI.removeDeviceFromSyncGroup(deviceGuid: guid)
+        }
+      case .syncChainDeleteConfirmation:
+        self.doIfConnected {
+          // Observers have to be removed before permanentlyDeleteAccount is called
+          // to prevent glitch in settings screen
+          self.syncAPI.removeAllObservers()
+          // Start  loading and diable navigation while delete action is happening
+          self.enableNavigationPrevention()
+          
+          // Permanently Delete action is called on brave-core side
+          self.syncAPI.permanentlyDeleteAccount { [weak self] status in
+            guard let self else { return }
+            
+            switch status {
+            case .throttled, .partialFailure, .transientError:
+              self.presentAlertPopup(for: .syncChainDeleteError)
+            default:
+              // Clearing local preferences and calling reset chain on brave-core side
+              self.syncAPI.leaveSyncGroup(includeObservers: false)
+            }
+            
+            self.disableNavigationPrevention()
+            self.navigationController?.popToRootViewController(animated: true)
+          }
+        }
+      default:
+        break
```

Comment:
Do we need this default? perhaps its better to list all enum values here, if its not too many of them

Reply:
Fix at https://github.com/brave/brave-ios/pull/7031/commits/2d7a1263608450affa1740f5fbdca40d253e7733